src/Search/InterestByCitySearch.php

```diff
@@ -0,0 +1,136 @@
+
+    /**
+     * @var ExploreSearch
+     */
+    protected $exploreSearch;
+
+    /**
+     * @var SearchRequest
+     */
+    protected $searchRequest;
+
+    public function __construct(ExploreSearch $exploreSearch = null, SearchRequest $searchRequest = null)
+    {
+        $this->searchRequest = $searchRequest ?: new SearchRequest();
+        $this->exploreSearch = $exploreSearch ?: new ExploreSearch($this->searchRequest);
+    }
+
+    /**
+     * @param SearchFilter $searchFilter
+     *
+     * @return InterestByRegionCollection
+     *
+     * @throws GoogleTrendsException
+     */
+    public function search(SearchFilter $searchFilter): ResultCollectionInterface
+    {
+        $token = $this->exploreSearch
+            ->search($searchFilter)
+            ->getInterestByRegionResult()
+            ->getToken();
+
+        $searchUrl = $this->buildQuery($searchFilter->withToken($token));
+
+        $responseDecoded = $this->searchRequest->search($searchUrl);
+
+        if (!isset($responseDecoded['default']['geoMapData'])) {
+            throw new GoogleTrendsException(
+                sprintf(
+                    'Invalid google response body "%s"...',
+                    substr(var_export($responseDecoded, true), 100)
+                )
+            );
+        }
+
+        $results = [];
+
+        foreach ($responseDecoded['default']['geoMapData'] ?? [] as $row) {
+            if (!isset($row['geoName'], $row['value'], $row['maxValueIndex'])) {
+                throw new GoogleTrendsException(
+                    sprintf(
+                        'Google compared geo list does not contain all keys. Only has: %s',
+                        implode(', ', array_keys($row))
+                    )
+                );
+            }
+            
+            $results[] = new InterestByCityResult(
+                $row['geoName'],
+                (int)($row['value'][0] ?? 0),
+                (int)$row['maxValueIndex'],
+                (bool)($row['hasData'] ?? false),
+                (string)($row['coordinates']['lat'] ?? false),
+                (string)($row['coordinates']['lng'] ?? false)
+            );
+        }
+
+        return new InterestByCityCollection($searchUrl, ...$results);
+    }
+
+    private function buildQuery(SearchFilter $searchFilter): string
+    {
+        $request = [
+            'geo' => [
+                'country' => $searchFilter->getLocation(),
+            ],
+            'comparisonItem' => [
+                [
+                    'time' => $searchFilter->getTime(),
+                    'complexKeywordsRestriction' => [
+                        'keyword' => [
+                            [
+                                'type' => 'BROAD',
+                                'value' => $searchFilter->getSearchTerm(),
+                            ],
+                        ],
+                    ]
+                ],
+            ],
+            'resolution' => 'CITY',
```

Comment:
@amarbhatia I saw the main difference between this class and the `InterestByRegionSearch` is this line and how to build result object https://github.com/gabrielfs7/google-trends/pull/27/files#diff-e036e59732258f52ead4508450d81e423fb6463e1383da577e735b1a159fcc33R72

I think we can add a new parameter on the `SearchFilter` (i.e resolution) and then instead of the `location` we would have 'getCountry' and 'getCity'.

This way we can avoid repeat code. What do you think?

Thanks for the PR. If you cannot do this changes, I will try when I have time :)

Reply:
sure, That would be great. It would be great if you did the changes and please add an option for wold wide filter as well.

Reply:
Hi @gabrielfs7, I have made the changes. I have removed InterestByCitySearch.php. I will be creating a PR today by EOD.

Reply:
hi @gabrielfs7, I have done everything in InterestByRegionSearch and removed the file InterestByCitySearch.php and also you can now search country wise as well. i have commited it to patch-3 branch. Please let me know if you need any help in uderstanding it and please forgive me if i made any mistakes.